clientServiceClient/api.go

```diff
@@ -0,0 +1,52 @@
+package clientServiceClient
+
+import (
+	"github.com/google/uuid"
+)
+
+type AdminListRequest struct {
+	NextToken int64 `json:"next_token"`
+	Limit     int   `json:"limit"`
+}
+
+type AdminListResponse struct {
+	Clients   []Client `json:"clients"`
+	NextToken int64    `json:"next_token"`
+}
+
+type AdminGetResponse struct {
+	Client
+}
+
+type ClientGetResponse struct {
+	Client
+}
+
+// ClientRegisterRequest captures the information sent to create a client.
+type ClientRegisterRequest struct {
+	RegistrationToken string `json:"token"`
+	Client            struct {
+		Name       string            `json:"name"`
+		Type       string            `json:"type"`
+		PublicKey  map[string]string `json:"public_key"`
+		SigningKey map[string]string `json:"signing_key,omitemtpy"`
+	} `json:"client"`
+}
+
+// ClientRegisterResponse sends back the JSON information of a client.
```

Comment:
nit: JSON is a redundant/junk food word in this comment. Suggest replacing it with something more information dense like "sends back the client information for a newly registered client"